lib/puma/thread_pool.rb

```diff
@@ -34,6 +38,11 @@ class ForceShutdown < RuntimeError
     # thread.
     #
     def initialize(name, options = {}, &block)
+      # below is for CI
+      @server = options.respond_to?(:user_options) ?
+        options.user_options.delete(:puma_server) :
+        options.delete(:puma_server)
```

Comment:
My personal opinion is that having to do this negates the micro optimisation gained from the core of this change (I don't think any lib code should exist just to satisfy CI). Could you link to the specific CI code that needs this? Is it just a naming conflict?

Reply:
As an aside: I would like (someone) to remove all `options = {}` in favor of explicit KWARGS where possible into the future.